modules/core/src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -184,20 +185,28 @@ private[kafka] final class KafkaConsumerActor[F[_], K, V](
       ref
         .modify { state =>
           val (newState, oldFetch) =
-            state.withFetch(partition, streamId, deferred)
+            state.withFetch(partition, streamId, partitionStreamId, deferred)
           (newState, (newState, oldFetch))
         }
         .flatMap {
-          case (newState, oldFetch) =>
+          case (newState, oldFetches) =>
             log(StoredFetch(partition, deferred, newState)) >>
-              oldFetch.fold(F.unit) { fetch =>
-                fetch.completeRevoked(Chunk.empty) >>
-                  log(RevokedPreviousFetch(partition, streamId))
+              oldFetches.fold(F.unit) { fetches =>
+                fetches
+                  .traverse(
+                    fetch =>
+                      fetch.completeRevoked(Chunk.empty) >>
+                        log(RevokedPreviousFetch(partition, streamId))
+                  )
+                  .void
               }
         }
 
     def completeRevoked =
-      deferred.complete((Chunk.empty, FetchCompletedReason.TopicPartitionRevoked))
+      deferred.complete((Chunk.empty, FetchCompletedReason.TopicPartitionRevoked)) >> ref.update {
```

Comment:
Shouldn't this `ref.update` happen in `revoked` instead of here?

Reply:
Oh yes, you are absolutely right!

Reply:
I pushed the comment related changes